### PR TITLE
Transformer.predict: do not broadcast to listeners

### PR DESCRIPTION
**Pull request by danieldk**
The output of a transformer is passed through in two different ways:

- Prediction: the data is passed through the `Doc._.trf_data` attribute.
- Training: the data is broadcast directly to the transformer's listeners.

However, the `Transformer.predict` method breaks the strict separation between training and prediction by also broadcasting transformer outputs to its listeners.

However, this breaks down when we are training a model with an unfrozen transformer when the transformer is also in `annotating_components`. The transformer will first (as part of its update step) broadcast the tensors and backprop function to its listeners. However, then when acting as an annotating component, it would immediately override its own output and clear the backprop function. As a result, gradients will not flow into the transformer.

This change removes the broadcast from the `predict` method. If a listener does not receive a batch, attempt to get the transformer output from the `Doc` instances. This makes it possible to train a pipeline with a frozen transformer.

This ports https://github.com/explosion/spaCy/pull/11385 to `spacy-transformers`. Alternative to #342.

### CODE REVIEW

File: spacy_transformers/layers/listener.py [hunk 1]
```diff
@@ -58,16 +58,32 @@ def verify_inputs(self, inputs):
...
+        # This might occur during training when the tok2vec layer is frozen / hasn't been updated.
+        # In that case, it should be set to "annotating" so we can retrieve the embeddings from the doc.
+        if model._batch_id is None:
+            outputs = []
+            for doc in docs:
+                if doc._.trf_data is None:
+                    # TODO move to errors.py
+                    raise ValueError(
+                        "If the embedding layer is not updated during training, include it in 'annotating components'"
+                    )
```
Comment (svlandeg) [inline spacy_transformers/layers/listener.py:70]: This can use E203 once https://github.com/explosion/spaCy/pull/11385 becomes available

File: spacy_transformers/layers/listener.py [hunk 2]
```diff
@@ -58,16 +58,32 @@ def verify_inputs(self, inputs):
 
 def forward(model: TransformerListener, docs, is_train):
     if is_train:
-        model.verify_inputs(docs)
-        return model._outputs, model.backprop_and_clear
+        # This might occur during training when the tok2vec layer is frozen / hasn't been updated.
```
Comment (svlandeg) [inline spacy_transformers/layers/listener.py:61]:
```suggestion
        # This might occur during training when the transformer layer is frozen / hasn't been updated.
```

File: spacy_transformers/layers/listener.py (hunk 1 above)
Comment (danieldk) [reply spacy_transformers/layers/listener.py:70]: Ok, I'll update this once spaCy 3.5.0 is out.

Comment (danieldk) [reply spacy_transformers/layers/listener.py]: Done 🎉 .
